### ISSUE TITLE
Warn about upcoming UA cut-off in Analytics settings

### ISSUE THREAD

**Issue by nfmohit** [google/site-kit-wp#6556]
## Feature Description

For sites that haven't yet connected GA4, a warning notification should be added to the Analytics `SettingsView` and `SettingsEdit` according to the [Figma designs](https://www.figma.com/file/0UCIE3hPeiu4NYdpAODASd/GA4-Updates?node-id=410%3A6399&t=w8y6XSruZn8Gwsaq-0). This should be built behind the `ga4Reporting` feature flag.

~For the warning notification, a new component, say `<NoticeBanner />`, can be created in a re-usable manner so that it can be used in other places.~ For now, use the `SettingsNotice` component, the new component will be made as part of #6702.

[Here](https://docs.google.com/document/d/14pCNmdPBqAJ-RfPhbHXfaMhGaPK-n-7F-6Sgujd0XCQ/edit#heading=h.8jga8k8r8va5) is the relevant section in the design doc.

Here is a screenshot for visual aid:
![image](https://user-images.githubusercontent.com/20284937/217533191-8cdf8f1c-0486-4038-8624-8f80805c89e8.png)

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance criteria

- For sites having GA4 not yet connected and the `ga4Reporting` feature flag enabled, the Analytics `SettingsEdit` and `SettingsView` should have a new warning notification added according to [this Figma design](https://www.figma.com/file/0UCIE3hPeiu4NYdpAODASd/GA4-Updates?node-id=410%3A6399&t=w8y6XSruZn8Gwsaq-0).
- While the contents of the warning notification should be identical to the Figma design, the UI should not be implemented. Instead, it should use the [existing notices UI](https://google.github.io/site-kit-wp/storybook/develop/?path=/story/global-notices--settings-info-notice-with-a-long-notice). The new design will be implemented as a part of #6702.
- Notice content: `Your current Universal Analytics property will stop collecting data on July 1, 2023`. (updated)
- The warning notification should have a `Learn more` CTA to the very right, which should navigate to the `ga4` documentation link URL.

## Implementation Brief

- In `assets/js/modules/analytics/components/settings/SettingsForm.js`:
  - Place a `<SettingsNotice />` component of type `warning` such that it appears on top of all settings (if #6547 is merged before this is started, this should appear underneath the setting added there).
  - The content of the notice should be according to the ACs.
  - Add a `Learn more` CTA, that should navigate the user to the GA4 documentation, obtainable using the `getDocumentationLinkURL()` selector from the `core/site` store, passing `ga4` to it.
- In `assets/js/modules/analytics/components/settings/SettingsView.js`:
  - Place another `<SettingsNotice />` component with the same structure as above on top all settings (if #6547 is merged before this is started, this should appear underneath the setting added there).
- The above notices should only appear if the site has the `ga4Reporting` feature flag enabled and GA4 is not connected.

### Test Coverage

* No new tests needed.

## QA Brief

- Enable the `ga4Reporting` feature flag.
- Ensure the GA4 is not connected. This can be simulated by deleting the `googlesitekit_analytics-4_settings` option from the `wp_options` table.
- Ensure the warning notice is displayed in the settings view and the settings form.
- Note that the new design isn't part of this ticket and will be implemented in #6702.

## Changelog entry

* Warn about upcoming Universal Analytics cut-off in Analytics settings.

**Comment by tofumatt** [comment]
ACs 👍🏻 

I think it makes more sense to create the component as part of this issue—it's always nicer to create things in real-world scenarios whenever possible. ~I'll mark the other two issues mentioned as blocked by this though, since they'll need to borrow this component/approach to work.~ You already did—thanks! 🙏🏻

**Comment by techanvil** [comment]
IB :white_check_mark:

**Comment by hussain-t** [comment]
@techanvil @nfmohit, can we use the `isGA4DashboardView` selector to conditionally render the warning instead of the `ga4Reporting` feature flag?

**Comment by nfmohit** [comment]
> @techanvil @nfmohit, can we use the `isGA4DashboardView` selector to conditionally render the warning instead of the `ga4Reporting` feature flag?

Hi Hussain! :wave:
I think it’d be wiser and simpler to use the `ga4Reporting` feature flag because the `isGA4DashboardView` selector uses a few additional conditions and is subject to change in [#6549](https://github.com/google/site-kit-wp/issues/6549).

**Comment by wpdarren** [comment]
### QA Update: ⚠️ 

@hussain-t I just have two observations.

1. The AC and FIgma designs say that the content should be:

`Your current Universal Analytics will stop collecting data on July 1, 2023`

For settings view and edit on my test site though I am seeing:

`Your current Universal Analytics property will stop collecting data on July 1, 2023`

In my opinion the text on my test site reads better than Figma, but wanted to highlight the difference. 

![image](https://user-images.githubusercontent.com/73545194/225176204-6cef207f-b399-4a7f-a72f-40cd304a3cd5.png)

What should the text be? 

2. It's very likely that this is expected or out of scope for this ticket but I wanted to highlight that when the `GA4Reporting` feature flag is enabled the GA4 activation banner does not appear on the dashboard. 

Should that be flagged separately, or, will it be fixed in another ticket? 

**QA Note:**  At the time of testing this ticket, the UI was to be excluded and is completely different than Figma design.

**Comment by hussain-t** [comment]
@wpdarren, regarding the text changes, @techanvil raised this suggestion and it was approved by @marrrmarrr 

As for:
>It's very likely that this is expected or out of scope for this ticket but I wanted to highlight that when the GA4Reporting feature flag is enabled the GA4 activation banner does not appear on the dashboard.

I could see the GA4 activation banner when the `ga4Reporting` feature flag is enabled, and the module is not connected. If it's not working or is an intermittent issue, can you create a separate issue?
